cpp/tests/streams/io/functions_test.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/io/detail/json.hpp>
+#include <cudf/io/json.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/types.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+
+#include <string>
+#include <vector>
+
+class FunctionsTest : public cudf::test::BaseFixture {};
+
+//.dtypes(std::vector<cudf::data_type>{dtype<int32_t>(), dtype<double>()})
+TEST_F(FunctionsTest, JSONreader)
+{
+  std::string data = "[1, 1.1]\n[2, 2.2]\n[3, 3.3]\n";
+  cudf::io::json_reader_options in_options =
+    cudf::io::json_reader_options::builder(cudf::io::source_info{data.data(), data.size()})
+      .dtypes(std::vector<cudf::data_type>{cudf::data_type{cudf::type_id::INT32},
+                                           cudf::data_type{cudf::type_id::FLOAT64}})
+      .lines(true)
+      .legacy(true);
+  cudf::io::table_with_metadata result = cudf::io::read_json(
+    in_options, cudf::get_default_stream(), rmm::mr::get_current_device_resource());
```

Comment:
The files in `cpp/tests/streams/*` need to use `cudf::test::get_default_stream()` to ensure they are also testing which stream is used. See docs: https://github.com/rapidsai/cudf/blob/e8cf0ebd517a9c81e294771a40c74ff5c4fb42da/cpp/include/cudf_test/default_stream.hpp#L27-L34
```suggestion
    in_options, cudf::test::get_default_stream(), rmm::mr::get_current_device_resource());
```
Please apply this change throughout the streams tests in this PR.

Reply:
Change applied to streams test `cudf::get_default_stream` -> `cudf::test::get_default_stream`